synapse/server.py

```diff
@@ -656,6 +657,10 @@ def get_replication_streams(self) -> Dict[str, Stream]:
     def get_federation_ratelimiter(self) -> FederationRateLimiter:
         return FederationRateLimiter(self.clock, config=self.config.rc_federation)
 
+    @cache_in_self
```

Comment:
I wonder if this should *not* be cached? Caching it means that different plugins could theoretically interfere with each other.

Reply:
well, they can probably do that anyway, if they so desire, since most of the functions are thin wrappers. :man_shrugging:

Reply:
True, probably not worth trying to protect against without a big effort.